docs/internals/layout_in_storage.rst

```diff
@@ -6,6 +6,9 @@ Layout of State Variables in Storage
 
 .. _storage-inplace-encoding:
 
+.. note::
+     The rules described in this section apply for both storage and transient storage data locations.
```

Comment:
I'd still add a few things:
- Maybe change title to "Layout of State Variables in Storage and Transient Storage"?
- Say that the layouts are completely independent, i.e. storage variables have zero effect on locations of transient storage variables and vice-versa. They can be safely interleaved.
- I'd mention that only value types are supported for transient storage.
- There's a note saying that we consider storage layout a part of the external interface. Is that still true for transient storage? If so, we should say that explicitly.
- Add an index entry for `transient storage`, next to `storage` above.

Reply:
Also, there's a note way below, saying:

> The JSON output format of a contract's storage layout is still considered experimental and is subject to change in non-breaking releases of Solidity.

I don't think it's true any more. When we once broke the layout accidentally in 0.8.x cycle, we considered it very breaking and released a new version a few days later to correct it.

Given that, I wonder if we should now say that about transient storage or is it automatically considered stable by the virtue of being tied to storage layout?

Reply:
I have applied all suggestions, but not sure about the notes regarding external interface. 
What do you think @ekpyron?

Reply:
TBH this is already wrong on `develop` so it's fine to merge the PR regardless and then fix it separately in a smaller doc-only PR.